### PR TITLE
refactor deleteMetric method

### PR DESCRIPTION
**Pull request by shabd67**
# Resolves

[CERES-472](https://jira.rax.io/browse/CERES-472)

# What

Refactor delete metric logic

# How

separate complex deletion logic of metrics by tenantId, metricName or metric Tag into smaller blocks of code to improve readability

### CODE REVIEW

File: src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java [hunk 1]
```diff
@@ -108,91 +109,44 @@ public MetricDeletionService(ReactiveCqlTemplate cqlTemplate,
...
+            deleteRawOrDownsampledEntries(dataTablesStatements
+                .downsampleDelete(granularity.getWidth()), tenant, timeSlot))
+        .then(deleteRawOrDownsampledEntries(dataTablesStatements.getRawDelete(),
+            tenant, timeSlot));
+  }
+
+
+  private Mono<Boolean> deleteMetadataByTenant(String tenant, Instant timeSlot) {
     //get series set hashes from data raw table by tenant and timeSlot
     Flux<String> seriesSetHashes = getSeriesSetHashFromRaw(tenant, timeSlot);
```
Comment (rohitsngh27) [inline src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java:130]: I think we need to first get all the seriesSetHashes as this method is called after the deletion from data tables and will result empty.

Comment (rohitsngh27) [reply src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java:130]: And also make sure to get series set hashes from the data table with highest granularity

**Review by rohitsngh27 — CHANGES_REQUESTED** [review]
More refactoring can be done. Need to add javadoc and combine methods which are having same use

File: src/main/java/com/rackspace/ceres/app/helper/MetricDeletionHelper.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  /**
+   * Delete series_set_hashes by tenant and seriesSetHash.
+   *
+   * @param tenant        the tenant
+   * @param seriesSetHash the series set hash
+   * @return the mono
+   */
+  public Mono<Boolean> deleteSeriesSetHashes(String tenant, String seriesSetHash) {
+    return cqlTemplate
+        .execute("DELETE FROM series_set_hashes WHERE tenant = ? AND series_set_hash = ?",
```
Comment (rohitsngh27) [inline src/main/java/com/rackspace/ceres/app/helper/MetricDeletionHelper.java:92]: Need to move this queries to constants I guess

Comment (shabd67) [reply src/main/java/com/rackspace/ceres/app/helper/MetricDeletionHelper.java:92]: these will be taken care in another PR https://github.com/racker/ceres/pull/43

**Review by rohitsngh27 — COMMENTED** [review]
Few minor change and need to write few more test cases as there can be many scenarios in this one.

File: src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java [hunk 3]
```diff
@@ -88,10 +91,9 @@ private void buildRawStatements(AppProperties appProperties) {
         + " WHERE " + TENANT + " = ?"
         + "  AND " + TIME_PARTITION_SLOT + " = ?"
         + " AND series_set_hash = ?";
-    rawGetHashQuery =
-        "SELECT series_set_hash FROM " + tableNameRaw(appProperties.getRawPartitionWidth())
-            + " WHERE " + TENANT + " = ?"
-            + "  AND " + TIME_PARTITION_SLOT + " = ?";
+    rawGetHashQuery = "SELECT series_set_hash FROM " + tableNameRaw(appProperties.getRawPartitionWidth())
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java:94]: You can declere these as separate private final static variables as strings with placeholders. Also I heard this is better for performance as well.

See example:

String s = "hello %s!";
s = String.format(s, "world");
assertEquals(s, "hello world!"); // should be true

File: src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java [hunk 4]
```diff
@@ -138,6 +140,14 @@ private void buildDownsampleStatements(DownsampleProperties downsampleProperties
                   + " AND series_set_hash = ?"
           );
         });
+
+    Granularity granularity = downsampleProperties.getGranularities()
+        .stream().max(Comparator.comparing(e -> e.getTtl())).get();
+
+    downsampledGetHashQuery = "SELECT series_set_hash FROM " + tableNameDownsampled(granularity.getWidth(),
```
Comment (matsboman) [inline src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java:147]: string with placeholder same comment

**Review by matsboman — CHANGES_REQUESTED** [review]

File: src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java (hunk 3 above)
Comment (shabd67) [reply src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java:94]: made changes in PR https://github.com/racker/ceres/pull/43

**Review by matsboman — DISMISSED** [review]
LGTM

**Review by harageth — APPROVED** [review]

**Review by rohitsngh27 — COMMENTED** [review]
LGTM. Let's wait for others to review.

**Review by rohitsngh27 — APPROVED** [review]

**Review by matsboman — APPROVED** [review]

**Pull request merged**